src/js/components/Select/Select.js

```diff
@@ -101,12 +101,20 @@ const Select = forwardRef(
         );
       return valueKey && valueKey.reduce ? value : applyKey(value, valueKey);
     }, [value, valueKey]);
+    // maintain master list of all options to reference when
+    // option values are selected from a filtered set of options
+    // (e.g. such as when searching)
+    const [optionsTotal, setOptionsTotal] = useState(options);
+    useEffect(() => {
+      if (options.length > optionsTotal.length) {
```

Comment:
I think this assumes that the caller is only reducing via search. For example, what if is a Select whose options depend on another form input element? For instance, if the Select contains "countries" whose options change based on another form input that selects from "continents", this would only reset optionsTotal when going from North America to Europe, but not vice-versa.

Reply:
Good point and example. Revisiting a concept I had explored prior.

Reply:
@ericsoderberghp I modified the approach to set `optionsTotal` to `initialOptions` only when searching. When not searching, use `options` like so:

```
const [search, setSearch] = useState();
const [initialOptions] = useState(options);
// if searching, use initialOptions
const optionsTotal = search ? initialOptions : options;
```